From d6afac962488157844345abe58e0352fa46f06b4 Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lotheac@iki.fi>
Date: Fri, 22 Jan 2016 18:46:27 +0200
Subject: [PATCH] don't pass -Y at all to the linker on illumos

the illumos linker can't understand the "-Y/lib"... form that f_check
generates, and -Wl cannot pass options that include commas
---
 f_check | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/f_check b/f_check
index e7e4688..4c9d81e 100644
--- a/f_check
+++ b/f_check
@@ -1,5 +1,7 @@
 #!/usr/bin/perl
 
+$hostos   = `uname -s | sed -e s/\-.*//`;    chop($hostos);
+
 #
 # 1. Not specified
 #   1.1 Automatically detect, then check compiler
@@ -272,8 +274,9 @@ if ($link ne "") {
 	}
 
 	if ($flags =~ /^\-Y/) {
+	    next if ($hostos eq 'SunOS');
 	    $linker_L .= "-Wl,". $flags . " ";
-	    }
+        }
 
 	if ($flags =~ /^\-rpath\@/) {
 	    $flags =~ s/\@/\,/g;
-- 
2.6.1

